src/cargo_ops/elaborate_workspace.rs

```diff
@@ -107,7 +107,7 @@ impl<'ela> ElaborateWorkspace<'ela> {
     fn find_contained_package(&self, name: &str) -> CargoResult<PackageId> {
         let root_path = self.workspace.root();
         for (pkg_id, pkg) in &self.pkgs {
-            if pkg.manifest_path().starts_with(root_path) && pkg.name().as_str() == name {
```

Comment:
Why `contains`? Is there a risk it'd confuse packages? (because `"foobar".contains("foo")`)

Reply:
I've found some packages end up as their repository name, which is often a variant of the package itself: `mydep-rs`, `rust-mydep` for `mydep`. I'll remove